src/main/scala/it/pagopa/interop/apigateway/api/impl/package.scala

```diff
@@ -0,0 +1,223 @@
+import scala.annotation.nowarn
+import scala.util.{Failure, Success, Try}
+
+package object impl extends SprayJsonSupport with DefaultJsonProtocol {
+  implicit val problemErrorFormat: RootJsonFormat[ProblemError] = jsonFormat2(ProblemError)
+  implicit val problemFormat: RootJsonFormat[Problem]           = jsonFormat5(Problem)
+
+  implicit val clientCredentialsResponseFormat: RootJsonFormat[ClientCredentialsResponse] = jsonFormat3(
+    ClientCredentialsResponse
+  )
+
+  implicit val purposeFormat: RootJsonFormat[Purpose]   = jsonFormat3(Purpose)
+  implicit val purposesFormat: RootJsonFormat[Purposes] = jsonFormat1(Purposes)
+
+  implicit val subscriberFormat: RootJsonFormat[Organization] = jsonFormat3(Organization)
+  implicit val eServiceFormat: RootJsonFormat[EService]       = jsonFormat4(EService)
+
+  implicit val agreementFormat: RootJsonFormat[Agreement]   = jsonFormat6(Agreement)
+  implicit val agreementsFormat: RootJsonFormat[Agreements] = jsonFormat1(Agreements)
+
+  implicit val attributeFormat: RootJsonFormat[Attribute] = jsonFormat3(Attribute)
+
+  implicit val attributeValidityStateFormat: RootJsonFormat[AttributeValidityState] = jsonFormat2(
+    AttributeValidityState
+  )
+
+  implicit val attributesFormat: RootJsonFormat[Attributes] = jsonFormat1(Attributes)
+
+  final val serviceErrorCodePrefix: String = "013"
+  final val defaultProblemType: String     = "about:blank"
+
+  def problemOf(httpError: StatusCode, error: ComponentError, defaultMessage: String = "Unknown error"): Problem =
+    Problem(
+      `type` = defaultProblemType,
+      status = httpError.intValue,
+      title = httpError.defaultMessage,
+      errors = Seq(
+        ProblemError(
+          code = s"$serviceErrorCodePrefix-${error.code}",
+          detail = Option(error.getMessage).getOrElse(defaultMessage)
+        )
+      )
+    )
+
+  private val descendingOffsetDateTimeOrdering: Ordering[OffsetDateTime] = implicitly[Ordering[OffsetDateTime]].reverse
+
+  implicit class EnrichedPurpose(private val purpose: PurposeManagementApiPurpose) extends AnyVal {
+    def toModel: Try[Purpose] =
+      purpose.versions
+        .sortBy(_.createdAt)(descendingOffsetDateTimeOrdering)
+        .find(p => p.state == PurposeVersionState.ACTIVE && p.state == PurposeVersionState.SUSPENDED)
+        .toTry(MissingActivePurposeVersion(purpose.id))
+        .map(version => Purpose(id = purpose.id, throughput = version.dailyCalls, state = version.state.toModel))
+  }
+
+  implicit class EnrichedPurposeVersionState(private val state: PurposeVersionState) extends AnyVal {
+    def toModel: PurposeState = state match {
+      case PurposeVersionState.ACTIVE    => PurposeState.ACTIVE
+      case DRAFT                         => PurposeState.DRAFT
+      case ARCHIVED                      => PurposeState.ARCHIVED
+      case WAITING_FOR_APPROVAL          => PurposeState.WAITING_FOR_APPROVAL
+      case PurposeVersionState.SUSPENDED => PurposeState.SUSPENDED
```

Comment:
let's use the same convention here 
```suggestion
      case PurposeVersionState.ACTIVE  => PurposeState.ACTIVE
      case PurposeVersionState.DRAFT  => PurposeState.DRAFT
      case PurposeVersionState.ARCHIVED => PurposeState.ARCHIVED
      case PurposeVersionState.WAITING_FOR_APPROVAL  => PurposeState.WAITING_FOR_APPROVAL
      case PurposeVersionState.SUSPENDED => PurposeState.SUSPENDED
```